### ISSUE TITLE
43.0.1 is not sampling correctly

### ISSUE THREAD

**Issue by mythmon** [mozilla/normandy#277]
The recipe "Out of Date Test 2" is attempting to sample English users running Firefox version 43.0.1 at a rate of 20%. We have evidence from telemetry that there are about 5.8 million people running 43.0.1. We have only gotten 300 offers for survey, which is significantly lower than 20% of 5.8 million, even accounting for locale.

The recipe's filter expression is:

```
(
   [normandy.userId, normandy.recipe.id]|stableSample(0.2)
   && normandy.locale in ['en-US', 'en-AU', 'en-CA', 'en-GB', 'en-NZ', 'en-ZA']
   && normandy.version == '43.0.1'
)
```

**Comment by mythmon** [comment]
My current theory is that 43.0.1 is some how not compatible with Normandy. I'm going to get a copy of 43.0.1 and test it's behavior.

**Comment by mythmon** [comment]
I've found the issue.

In Firefox 43.0.1 in my local tests, the request to `/api/v1/action/:id/` is made with the header `Accept: text/html`. In Firefox 52 (my local nightly), the request is made with the correct header `Accept: application/json`. It seems that the default content type for XHR changed between 43 and 52. When an `Accept: text/html` is sent, the server correctly responded with an HTML page containing the API response, instead of the JSON response the client expects.

This wasn't an issue until recently. Before we deployed v18 (around September 21st), we had Nginx forcing the Accept header to be `application/json` because of [bug 1293469](https://bugzilla.mozilla.org/show_bug.cgi?id=1293469). We ignored what the client sent, so Firefox's default behavior didn't matter.

The fix for this is pretty easy. We need to fix the api call to load action data to send the proper headers. That happens in [fetchAction](https://github.com/mozilla/normandy/blob/master/client/selfrepair/self_repair_runner.js#L42-L58), which does not set any extra headers on the request. For comparison, in [fetchRecipes](https://github.com/mozilla/normandy/blob/master/client/selfrepair/self_repair_runner.js#L75-L85), the header is correctly added.
